astropy/nddata/utils.py

```diff
@@ -456,3 +469,267 @@ def block_replicate(data, block_size, conserve_sum=True):
+        >>> from astropy.nddata.utils import Cutout
+        >>> data = np.arange(20.).reshape(5, 4)
+        >>> c1 = Cutout(data, (2, 2), (3, 3))
+        >>> print(c1.data)
+        [[  5.   6.   7.]
+         [  9.  10.  11.]
+         [ 13.  14.  15.]]
+
+        >>> print(c1.center_large)
+        (2.0, 2.0)
+        >>> print(c1.center_small)
+        (1.0, 1.0)
+        >>> print(c1.origin_large)
+        (1, 1)
+
+        >>> c2 = Cutout(data, (0, 0), (3, 3))
+        >>> print(c2.data)
+        [[ 0.  1.]
+         [ 4.  5.]]
+
+        >>> c3 = Cutout(data, (0, 0), (3, 3), mode='partial')
+        >>> print(c3.data)
+        [[ nan  nan  nan]
+         [ nan   0.   1.]
+         [ nan   4.   5.]]
+        """
+
+        if isinstance(position, SkyCoord):
+            if wcs is None:
+                raise ValueError('wcs must be input if position is a '
+                                 'SkyCoord')
+            position = skycoord_to_pixel(position, wcs, mode='all')
+
+        if mode not in ['partial', 'trim', 'strict']:
+            raise ValueError("Valid modes are 'partial', 'trim', and "
+                             "'strict'.")
+
+        data = np.asanyarray(data)
+        self.data, self.input_position_small  = extract_array(data, shape,
+            position, mode=mode, fill_value=fill_value, return_position=True)
+
+        slices_large, slices_small = overlap_slices(data.shape, shape,
+                                                    position, mode=mode)
+        self.slices_large = slices_large
+        self.slices_small = slices_small
+
+        self.shape = self.data.shape
+        self.input_position_large = position
+        self.shape_input = shape
+
+        (self.ymin_large, self.xmin_large,
+         self.ymax_large, self.xmax_large) = self.bbox_large
+
+        (self.ymin_small, self.xmin_small,
+         self.ymax_small, self.xmax_small) = self.bbox_small
+
+        # the true origin pixel of the cutout array, including any
+        # filled cutout values
+        self._origin_large_true = (self.origin_large[1] -
+                                   self.slices_small[1].start,
+                                   self.origin_large[0] -
+                                   self.slices_small[0].start)
+
+        if wcs is not None:
+            self.wcs = copy.deepcopy(wcs)
+            self.wcs.wcs.crpix -= self._origin_large_true
+        else:
+            self.wcs = None
+
+    def to_large(self, position):
+        """
+        Calculate the pixel position in the large array corresponding to
+        the input pixel position in the cutout array.
+
+        Parameters
+        ----------
+        position : tuple
+            The ``(y, x)`` pixel position in the cutout array.
+
+        Returns
+        -------
+        position : tuple
+            The corresponding ``(y, x)`` pixel position in the large
+            array.
+        """
+        return tuple(position[i] + self.origin_large[i] for i in [0, 1])
```

Comment:
The `to_large` and `from_large` one-line methods are the only ones not covered by unit tests.
I think they are covered by docs tests, but those don't run for Python 3, so maybe useful to add a unit test for those?